diff --git a/src/fido/cbor_make_credential.c b/src/fido/cbor_make_credential.c
index 5875af7..1c0036b 100644
--- a/src/fido/cbor_make_credential.c
+++ b/src/fido/cbor_make_credential.c
@@ -505,6 +505,14 @@ int cbor_make_credential(const uint8_t *data, size_t len) {
 #endif
             }
         }
+        else if (!(flags & FIDO2_AUT_FLAG_UP)) {
+            if (check_user_presence() == false) {
+                CBOR_ERROR(CTAP2_ERR_OPERATION_DENIED);
+            }
+#ifndef ENABLE_EMULATION
+            button_pressed = phy_data.up_btn != 0;
+#endif
+        }
         flags |= FIDO2_AUT_FLAG_UP;
         clearUserPresentFlag();
         clearUserVerifiedFlag();
