diff --git a/src/fido/cbor_get_info.c b/src/fido/cbor_get_info.c
index 5410143..ca8ec8e 100644
--- a/src/fido/cbor_get_info.c
+++ b/src/fido/cbor_get_info.c
@@ -73,7 +73,7 @@ int cbor_get_info(void) {
     CborError error = CborNoError;
     uint8_t enc_identifier[DEV_STATE_SIZE] = { 0 }, enc_cred_store_state[DEV_STATE_SIZE] = { 0 };
     cbor_encoder_init(&encoder, ctap_resp->init.data + 1, CTAP_MAX_CBOR_PAYLOAD, 0);
-    uint8_t lfields = 20;
+    uint8_t lfields = 21;
     file_t *ef_ee_ea = file_search_by_fid(EF_EE_DEV_EA, NULL, SPECIFY_EF);
     bool enterprise_profile = ((get_opts() & FIDO2_OPT_EA) && file_has_data(ef_ee_ea));
 #ifndef ENABLE_EMULATION
@@ -154,6 +154,11 @@ int cbor_get_info(void) {
     CBOR_CHECK(cbor_encode_uint(&mapEncoder, 0x08));
     CBOR_CHECK(cbor_encode_uint(&mapEncoder, MAX_CRED_ID_LENGTH)); // MAX_CRED_ID_MAX_LENGTH
 
+    CBOR_CHECK(cbor_encode_uint(&mapEncoder, 0x09));
+    CBOR_CHECK(cbor_encoder_create_array(&mapEncoder, &arrayEncoder, 1));
+    CBOR_CHECK(cbor_encode_text_stringz(&arrayEncoder, "usb"));
+    CBOR_CHECK(cbor_encoder_close_container(&mapEncoder, &arrayEncoder));
+
     CBOR_CHECK(cbor_encode_uint(&mapEncoder, 0x0A));
 
     uint8_t curves = 3;
